wgpu/src/util/mod.rs

```diff
@@ -155,24 +155,43 @@ impl std::ops::Deref for DownloadBuffer {
 ///
 /// # Examples
 ///
-/// ``` no_run
+/// ```no_run
 /// # use std::path::PathBuf;
+/// use wgpu::PipelineCacheDescriptor;
 /// # let adapter_info = todo!();
-/// let cache_dir: PathBuf = PathBuf::new();
+/// # let device: wgpu::Device = todo!();
+/// let cache_dir: PathBuf = unimplemented!("Some reasonable platform-specific cache directory for your app.");
 /// let filename = wgpu::util::pipeline_cache_key(&adapter_info);
-/// if let Some(filename) = filename {
-///     let cache_file = cache_dir.join(&filename);
-///     let cache_data = std::fs::read(&cache_file);
-///     let pipeline_cache: wgpu::PipelineCache = todo!("Use data (if present) to create a pipeline cache");
+/// let (pipeline_cache, cache_file) = if let Some(filename) = filename {
+///     let cache_path = cache_dir.join(&filename);
+///     // If we failed to read the cache, for whatever reason, treat the data as lost.
+///     // In a real app, we'd probably avoid caching entirely unless the error was "file not found".
+///     let cache_data = std::fs::read(&cache_path).ok();
+///     let pipeline_cache = unsafe {
+///         device.create_pipeline_cache(&PipelineCacheDescriptor {
+///             data: cache_data.as_deref(),
+///             label: None,
+///             fallback: true
+///         })
+///     };
+///     (Some(pipeline_cache), Some(cache_path))
```

Comment:
nit: might as well do an `Option<(cache, path)>` instead of a `(Option<cache>, Option<path>)`.

Reply:
ah, nevermind.